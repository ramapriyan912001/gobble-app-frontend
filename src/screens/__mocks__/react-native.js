// import * as ReactNative from 'react-native';

// jest.doMock('react-native', () => {
//   return Object.setPrototypeOf(
//     {
//       Dimensions: {
//         get: () => ({width: 414, height: 896}),
//       },
//       Platform: {
//         ...ReactNative.Platform,
//         OS: 'ios',
//         select: jest.fn((selector) => selector),
//       },
//       NativeModules: {
//         ...ReactNative.NativeModules,
//         SettingsManager: {
//           settings: {
//             AppleLocale: 'en_US',
//           },
//         },
//         I18nManager: {
//           localeIdentifier: 'en_US',
//         },
//       },
//     },
//     ReactNative,
//   );
// });
