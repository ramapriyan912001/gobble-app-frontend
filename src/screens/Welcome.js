import React, {useEffect, useState} from 'react'
import {Text, SafeAreaView, Image, TouchableOpacity, Alert, LogBox} from 'react-native'
import {imageStyles, inputStyles, containerStyles, profileStyles, buttonStyles} from '../styles/LoginStyles'
import deviceStorage from '../services/deviceStorage'
import {API} from '../api'
import jwt from 'expo-jwt';
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { fetchAuthUser, clearData } from '../redux/actions/actions'

LogBox.ignoreLogs(['Unhandled promise rejection: Error: Native splash screen is already hidden. Call this method before rendering any view.'])

//TODO: This shouldn't be part of a stack navigator. Users can swipe left from profile page and see this.

export function Welcome(props) {
    // async function postDataAsync() {
    //     await deviceStorage
    //         .loadJWT()
    //         .then(async token => {
    //             let payload = 
    //             jwt
    //             .decode(
    //             token, // the token
    //             'iuewifu398b', // the secret - find safer place to store
    //             { timeSkew: 60 }
    //             );

    //             const config = {
    //                 headers: { Authorization: `Bearer ${token}`, 'Content-Type': 'application/json'}
    //             };

    //             await API
    //             .put( 
    //                 `users/${payload.userID}/lastseen`,
    //                 {
    //                     lastSeen: Date.now(),
    //                     method: 'PUT'
    //                 }
    //                 ,config
    //             )
    //             .then(res => {
    //                 if (!res.data.success) {
    //                     Alert.alert('Could not update user details - Network Problem');
    //             }})
    //             .catch(err => console.log(err.response.data.message));
    //         })
    //         .catch(err => console.log(err));
    //     };
        
    // useEffect(() => {
    //     postDataAsync();
    // },[]);

    return (
        <SafeAreaView style={containerStyles.container}>
            <Text style={inputStyles.headerText}>Welcome to</Text>
            <Image style={imageStyles.gobbleImage} source = {require('../images/gobble.png')}/>
            <Text style={profileStyles.profileField}>Gobble is the app that connects you to the world over a simple meal</Text>
            <Text style={profileStyles.profileField}>Use the Matches Page to search for a Match based on your own criteria</Text>
            <Text style={profileStyles.profileField}>Once you find a match, you can chat with them from your chats screen!</Text>
            <Text style={profileStyles.profileField}>After you confirm the details with your match, it's Gobble time!</Text>
            <Text style={profileStyles.profileField}>Tap Next to Start</Text>
            <TouchableOpacity style={buttonStyles.loginButton}
                        onPress={()=> {props.navigation.navigate('Profile')}}>
                <Text style={buttonStyles.loginButtonText}>Next</Text>
            </TouchableOpacity>
        </SafeAreaView>
        );
};

const mapStateToProps = (store) => ({
    currentUserData: store.userState.currentUserData,
    loggedIn: store.userState.loggedIn,
    isAdmin: store.userState.isAdmin,
})
const mapDispatchProps = (dispatch) => bindActionCreators({ fetchAuthUser, clearData }, dispatch);
export default connect(mapStateToProps, mapDispatchProps)(Welcome);


