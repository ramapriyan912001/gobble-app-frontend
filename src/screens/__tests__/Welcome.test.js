import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Welcome from '../Welcome';
import { expect, it, jest } from '@jest/globals';

// const gobbleImg = jest.mock('../../images/gobble.png');

describe('<Welcome />', () => {
    it('renders default elements', () => {
        const { getByText, getByTestId } = render(<Welcome/>);
        getByTestId('GobbleImage');
        getByText('Connecting People Over Food.');
        getByText('Already have an Account?');
        getByText('Sign Up');
    });

    it('can navigate to login', () => {
        const navigateMock = jest.fn();
        const { getByTestId } = render(<Welcome navigation={{ navigate: navigateMock }}/>);

        fireEvent.press(getByTestId('ToLoginButton'));
        expect(navigateMock).toBeCalledWith('Login');
    });

    it('can navigate to register', () => {
        const navigateMock = jest.fn();
        const { getByTestId } = render(<Welcome navigation={{ navigate: navigateMock }}/>);

        fireEvent.press(getByTestId('ToRegisterButton'));
        expect(navigateMock).toBeCalledWith('RegisterNavigator');
    });
});