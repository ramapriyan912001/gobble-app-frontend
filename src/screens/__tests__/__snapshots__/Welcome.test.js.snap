// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Welcome /> should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
      },
      Object {
        "backgroundColor": "#242C40",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": "40%",
        "marginTop": "0%",
        "width": "70%",
      }
    }
    testID="GobbleImage"
  />
  <View
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 38,
            "fontWeight": "bold",
            "marginBottom": "0%",
            "marginLeft": "1%",
            "textAlign": "left",
          },
          Object {
            "color": "#ffe",
          },
        ]
      }
    >
      Connecting People Over Food.
    </Text>
  </View>
  <View
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginVertical": "6%",
          "opacity": 1,
        }
      }
      testID="ToRegisterButton"
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
            },
            Object {
              "color": "#ffe",
            },
          ]
        }
      >
        Sign Up
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="ToLoginButton"
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
            },
            Object {
              "color": "#ffe",
            },
          ]
        }
      >
        Already have an Account?
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
